### PR TITLE
Reworks the KOTS config to be more opinionated

### PR DESCRIPTION
**Pull request by crdant**
TL;DR
-----

Simplifies the KOTS admin screen by asserting more opinions

Details
-------

Makes the following updates to the KOTS config screen in hopes
of simplfiying it per #45 :

1. Avoids asking about NGINX. It's used for all deployments which
   allows for always specifying `ClusterIP` as the service type
   for the Slackernews service. This means only one prompt for
   the service type and one less option around NGINX.

2. Uses file upload for the TLS certificate fields. Also makes it
   explicit whether the install generates the certificate or the
   user supplies it. I used this change as an opportunity to move
   the generation of a cert into the `HelmChart` object.

3. Migrates the admin users and domain to a "Application Core"
   configuration group. I don't love the name, so would love
   additional suggestions there.

4. Breaks out TLS and Ingress options into their own groups for
   readablity. A bit of a personal preference to do it this way
   (kind of like the upload vs. textarea thing for the certs) but
   I feel like it breaks up the page a bit more and makes it
   friendlier.

5. Moves the database config to the bottom. It felt odd that the
   first thing I was asked for was how I wanted to handle the
   database. So I tried this instead.

The goal of these change is twofold: to improve the UX for a user
running an install with KOTS, and to show our vendors how the KOTS
config can be used to assert their opinons on configuration rather
than just parroting the Helm values.

### CODE REVIEW

**Comment by crdant** [comment]
Here's a quick loom of what this looks like before/after and my thinking.

https://www.loom.com/share/36eb4a4bdf7f49f6bb04bf9d97bc6d17

**Review by marccampbell — APPROVED** [review]

**Comment by crdant** [comment]
Couple of additional tweaks:

1. Added validation
2. Fixed broken `HelmChart` kind
